lib/plenario/schemas/unique_constraint.ex

```diff
@@ -9,6 +9,7 @@ defmodule Plenario.Schemas.UniqueConstraint do
 
   alias Plenario.Schemas.UniqueConstraint
 
+  @derive {Poison.Encoder, only: [:name, :field_ids]}
```

Comment:
With this and the the `Virtual*Field`s, I would rather see the names of the referenced fields instead of their `id`s. An argument can be made that clients can programmatically get the names by using the IDs, but that makes this difficult to visually grep through.

We could include IDs for all things, but I feel strongly that we should make the extra effort to include the names of the fields here and in the virtual fields.

```elixir
defimpl Poison.Encoder, for: Plenario.Schemas.UniqueConstraint do
  def encode(uc, _opts) do
    field_names = Repo.all(from d in DataSetFieldActions, where: d.id in ^uc.field_ids, select: d.name)
    ...
  end
end
```

Reply:
I'm a little worried about the potential performance cost of an extra query for each unique constraint, but I agree that it would be much more useful to clients. I'm going to update this and the next comment accordingly and if the endpoints become too slow we can come back and reassess.